src/nvim/main.c

```diff
@@ -724,6 +728,34 @@ static void init_locale(void)
 }
 #endif
 
+/// Initialize the client
+static bool client_init(int remote, char *server_addr, int argc, char **argv){
```

Comment:
`client_init` is not a good name, as this also invokes the clients and exits. Also instead of `if (remote) {`, you could avoid calling the function when not used, and reduce the nesting.

Reply:
@bfredl I put it that way because I thought there will be other things needed to be done inside `client_init` other than handling the remote client. I changed the name to `handle_remote_client` and changed the return type to `void` from `bool` as it was returning nothing in the last commit. I am looking at nvm implementation of the `wait` variants and the tests.